src/shared/ui/components/TransactionsList.vue

```diff
@@ -0,0 +1,154 @@
+<script setup lang="ts">
+import { computed, reactive, watch } from 'vue';
+import { useTable } from '@/shared/lib/table';
+import * as http from '@/shared/api';
+import { objectOmit } from '@vueuse/shared';
+import {
+  type filterTransactionsModel as ftm,
+  InstructionTypeFilter,
+  TransactionStatusFilter,
+} from '@/features/filter-transactions';
+import type { AccountId } from '@/shared/api/schemas';
+import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import { accountInstructionsAdaptiveOptions } from '@/features/filter-transactions/adaptive-options';
+import InstructionsTable from '@/shared/ui/components/InstructionsTable.vue';
+import TransactionsTable from '@/shared/ui/components/TransactionsTable.vue';
+import { useWindowSize } from '@vueuse/core';
+
+const { handleUnknownError } = useErrorHandlers();
+const props = withDefaults(
+  defineProps<{
+    showInstructions?: boolean
+    filterBy?: { kind: 'authority', id: AccountId } | { kind: 'block', block: number } | null
+  }>(),
+  { filterBy: null }
+);
+
+const transactionsTable = useTable(http.fetchTransactions, { reversed: true });
+const instructionsTable = useTable(http.fetchInstructions);
+
+const listState = reactive({
+  status: null,
+  authority: props.filterBy?.kind === 'authority' ? props.filterBy.id.toString() : '',
+  kind: '' as ftm.TabInstructions,
+  block: props.filterBy?.kind === 'block' ? props.filterBy.block : '',
```

Comment:
Improper use of `reactive`: it ignores changes to `filterBy`.

I anticipate you saying: "Yes, it is okay. `filterBy` is not changing in the application anyway, it is only set on the first component load"

True. Please make it properly anyway.

Also, don't set empty string for absent values. Better to omit the fields entirely.